src/openrct2/peep/staff.c

```diff
@@ -502,11 +502,6 @@ void game_command_fire_staff_member(sint32 *eax, sint32 *ebx, sint32 *ecx, sint3
             *ebx = MONEY32_UNDEFINED;
             return;
         }
-        if (peep->state == PEEP_STATE_PICKED) {
-            *ebx = MONEY32_UNDEFINED;
-            return;
-        }
-        remove_peep_from_ride(peep);
```

Comment:
Are you sure this is not going to have side effects? Especially the removal of the `remove_peep_from_ride()` call? (I did check that function and I think it should be okay, but any particular reason you removed it?)

Reply:
`remove_peep_from_ride` is the first function called in `peep_sprite_remove` so it's okay.